standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/datasource/HikariCPDataSourceProvider.java

```diff
@@ -72,6 +72,17 @@ public DataSource create(Configuration hdpConfig, int maxPoolSize) throws SQLExc
       config.setPoolName(poolName);
     }
 
+    // It's kind of a waste to create a fixed size connection pool as same as the TxnHandler#connPool,
+    // TxnHandler#connPoolMutex is mostly used for MutexAPI that is primarily designed to
+    // provide coarse-grained mutex support to maintenance tasks running inside the Metastore,
+    // add minimumIdle=2 and idleTimeout=5min to the pool, so that the connection pool can retire
+    // the idle connection aggressively, this will make Metastore more scalable especially if
+    // there is a leader in the warehouse.
+    if ("mutex".equals(poolName)) {
```

Comment:
Could we please extract the string literal into constants?

Reply:
This literal is used only in creating data source, so in my point of view we can use it directly.

Reply:
👍